src/main/java/io/vertx/core/eventbus/MessageConsumer.java

```diff
@@ -90,6 +90,11 @@
    */
   void completionHandler(Handler<AsyncResult<Void>> completionHandler);
 
+  /**
+   * Like {@link #completionHandler(Handler)} but returns a {@code Future} of the asynchronous result.
+   */
+  Future<Void> registrationCompletion();
+
```

Comment:
we also need a `registrationCompletion(Handler<AsyncResult<Void>>)` method to go along with this

Reply:
I think after this round i understand your intention :)